### ISSUE TITLE
Lost configure file

### ISSUE THREAD

**Issue by smft** [uturuncoglu/RegESM#17]
configure file is not in the master branch. Are there any easier way to compile this thing without configure file ？

**Comment by uturuncoglu** [comment]
Actually, configuration file will be generated by issuing ./bootstrap command. So, after creating configure file you could continue to the installation with regular ./configuration, make, make install chain. This was not clear in the documentation and i fix it now. Thanks for pointing this out. Let me know if you have still problem and i'll keep open this issue.

**Comment by smft** [comment]
Yeah, already find, thanks a lot. But here come out an other bug.
ifort -Iutil -I/opt/esmf-7.1.0-ifort-icpc/mod/modO/Linux.intel.64.intelmpi.default -I/opt/esmf-7.1.0-ifort-icpc/include -I/opt/netcdf-4.1.3-icc-ifort-icpc/include -I/opt/hdf5-1.8.19-icc-ifort-icpc/include -I/opt/zlib-1.2.8-icc/include -I/opt/xerces-3.2.1-icc-icpc/include -I/HGST_SAS_1/RegESM/regcm-release-450/Main -I/HGST_SAS_1/RegESM/regcm-release-450/Main/mpplib -I/HGST_SAS_1/RegESM/regcm-release-450/Main/batslib -I/HGST_SAS_1/RegESM/regcm-release-450/Share -I/HGST_SAS_1/RegESM/roms-release-809/Build     -O3 -fp-model precise -fp-model source -ip -g -c mod_esmf_atm.F90
mod_esmf_atm.F90(2274): error #6580: Name in only-list does not exist or is not accessible.   [HTSIG_S]
      use mod_hgt, only : htsig_s, nonhydrost_s
--------------------------^
mod_esmf_atm.F90(2274): error #6580: Name in only-list does not exist or is not accessible.   [NONHYDROST_S]
      use mod_hgt, only : htsig_s, nonhydrost_s
-----------------------------------^
mod_esmf_atm.F90(2275): error #6580: Name in only-list does not exist or is not accessible.   [INTLINREGZ]
      use mod_vertint, only : intlinregz
------------------------------^
mod_esmf_atm.F90(2277): error #6580: Name in only-list does not exist or is not accessible.   [EXPORTFIELDS3D]
      use mod_update, only : exportFields, exportFields3d
-------------------------------------------^
mod_esmf_atm.F90(2677): error #6404: This name does not have a type, and must have an explicit type.   [EXPORTFIELDS3D]
        call htsig_s(exportFields3d%t, hzvar, exportFields%psfc*d_100,&
---------------------^
mod_esmf_atm.F90(2677): error #6460: This is not a field name that is defined in the encompassing structure.   [T]
        call htsig_s(exportFields3d%t, hzvar, exportFields%psfc*d_100,&
------------------------------------^
mod_esmf_atm.F90(2708): error #6460: This is not a field name that is defined in the encompassing structure.   [Q]
        call intlinregz(zvar, exportFields3d%q, hzvar, sigma,           &
---------------------------------------------^
mod_esmf_atm.F90(2712): error #6460: This is not a field name that is defined in the encompassing structure.   [U]
        call intlinregz(zvar, exportFields3d%u, hzvar, sigma,           &
---------------------------------------------^
mod_esmf_atm.F90(2716): error #6460: This is not a field name that is defined in the encompassing structure.   [V]
        call intlinregz(zvar, exportFields3d%v, hzvar, sigma,           &
---------------------------------------------^
mod_esmf_atm.F90(2720): error #6460: This is not a field name that is defined in the encompassing structure.   [W]
        call intlinregz(zvar, exportFields3d%w, hzvar, sigma,           &
---------------------------------------------^
mod_esmf_atm.F90(2724): error #6460: This is not a field name that is defined in the encompassing structure.   [CLDFRC]
        call intlinregz(zvar, exportFields3d%cldfrc, hzvar, sigma,      &
---------------------------------------------^
mod_esmf_atm.F90(2728): error #6460: This is not a field name that is defined in the encompassing structure.   [CLDLWC]
        call intlinregz(zvar, exportFields3d%cldlwc, hzvar, sigma,      &
---------------------------------------------^


I've already checked in the regcm-4.5.0 (successfully compiled), and there actually No "htsig_s","nonhydrost_s","intlinregz","exportFields3d" in ${regcm_path}/Share/mod_hgt.f90

I'm using intel 2018 compiler with netCDF4.1.3,HDF5-1.8.19,esmf-7.1.0

**Comment by smft** [comment]
In the README.md, it is said "it needs extra patch for co-processing", but the patch cannot be found.

**Comment by uturuncoglu** [comment]
Yes. For co-processing, there are some extra modification in atmosphere model to provide also three-dimensional fields (interpolated to z-coordinates from sigma). Unfortunately, we are not add those modifications directly to RegCM and now we have our own version and only patch. We are currently testing WRF as new atmosphere model component. The interface is ready but it is still in testing phase. Anyway, are you planing to use co-processing? We are currently working on a paper for GMD and it will be submitted in a couple of months. If not please comment out all code related with 3d fields and also vertical interpolation. If you don't activate co-processing component, the rest of the driver works just fine. Let me know if you need help.

**Comment by smft** [comment]
That is very nice of you!

Looking forward to your WRF version.

**Comment by BGM-USD2020** [comment]
> Yeah, already find, thanks a lot. But here come out an other bug.
> ifort -Iutil -I/opt/esmf-7.1.0-ifort-icpc/mod/modO/Linux.intel.64.intelmpi.default -I/opt/esmf-7.1.0-ifort-icpc/include -I/opt/netcdf-4.1.3-icc-ifort-icpc/include -I/opt/hdf5-1.8.19-icc-ifort-icpc/include -I/opt/zlib-1.2.8-icc/include -I/opt/xerces-3.2.1-icc-icpc/include -I/HGST_SAS_1/RegESM/regcm-release-450/Main -I/HGST_SAS_1/RegESM/regcm-release-450/Main/mpplib -I/HGST_SAS_1/RegESM/regcm-release-450/Main/batslib -I/HGST_SAS_1/RegESM/regcm-release-450/Share -I/HGST_SAS_1/RegESM/roms-release-809/Build -O3 -fp-model precise -fp-model source -ip -g -c mod_esmf_atm.F90
> mod_esmf_atm.F90(2274): error #6580: Name in only-list does not exist or is not accessible. [HTSIG_S]
> use mod_hgt, only : htsig_s, nonhydrost_s
> --------------------------^
> mod_esmf_atm.F90(2274): error #6580: Name in only-list does not exist or is not accessible. [NONHYDROST_S]
> use mod_hgt, only : htsig_s, nonhydrost_s
> -----------------------------------^
> mod_esmf_atm.F90(2275): error #6580: Name in only-list does not exist or is not accessible. [INTLINREGZ]
> use mod_vertint, only : intlinregz
> ------------------------------^
> mod_esmf_atm.F90(2277): error #6580: Name in only-list does not exist or is not accessible. [EXPORTFIELDS3D]
> use mod_update, only : exportFields, exportFields3d
> -------------------------------------------^
> mod_esmf_atm.F90(2677): error #6404: This name does not have a type, and must have an explicit type. [EXPORTFIELDS3D]
> call htsig_s(exportFields3d%t, hzvar, exportFields%psfc_d_100,& ---------------------^ mod_esmf_atm.F90(2677): error #6460: This is not a field name that is defined in the encompassing structure. [T] call htsig_s(exportFields3d%t, hzvar, exportFields%psfc_d_100,&
> ------------------------------------^
> mod_esmf_atm.F90(2708): error #6460: This is not a field name that is defined in the encompassing structure. [Q]
> call intlinregz(zvar, exportFields3d%q, hzvar, sigma, &
> ---------------------------------------------^
> mod_esmf_atm.F90(2712): error #6460: This is not a field name that is defined in the encompassing structure. [U]
> call intlinregz(zvar, exportFields3d%u, hzvar, sigma, &
> ---------------------------------------------^
> mod_esmf_atm.F90(2716): error #6460: This is not a field name that is defined in the encompassing structure. [V]
> call intlinregz(zvar, exportFields3d%v, hzvar, sigma, &
> ---------------------------------------------^
> mod_esmf_atm.F90(2720): error #6460: This is not a field name that is defined in the encompassing structure. [W]
> call intlinregz(zvar, exportFields3d%w, hzvar, sigma, &
> ---------------------------------------------^
> mod_esmf_atm.F90(2724): error #6460: This is not a field name that is defined in the encompassing structure. [CLDFRC]
> call intlinregz(zvar, exportFields3d%cldfrc, hzvar, sigma, &
> ---------------------------------------------^
> mod_esmf_atm.F90(2728): error #6460: This is not a field name that is defined in the encompassing structure. [CLDLWC]
> call intlinregz(zvar, exportFields3d%cldlwc, hzvar, sigma, &
> ---------------------------------------------^
> 
> I've already checked in the regcm-4.5.0 (successfully compiled), and there actually No "htsig_s","nonhydrost_s","intlinregz","exportFields3d" in ${regcm_path}/Share/mod_hgt.f90
> 
> I'm using intel 2018 compiler with netCDF4.1.3,HDF5-1.8.19,esmf-7.1.0

Hi! Qi, have you solved this problem already? if yes, could you please share your solution with me?

**Comment by uturuncoglu** [comment]
@BGM-USD2020 if you don't need to use latest code please download version 1.0 from https://github.com/uturuncoglu/RegESM/releases/tag/1.0.0. The WRF coupling and in-situ visualization are added after that release.

**Comment by mngbyz** [comment]
Hi! I got an error with regcm. Tried version 1.0.0. But another compilation error appeared

mod_esmf_ocn_roms.F90 (111): error # 7062: The characteristics of dummy argument 2 of the associated actual procedure differ from the characteristics of dummy argument 2 of the dummy procedure. [OCN_SETCLOCK]
                                 specRoutine = OCN_SetClock, rc = rc)
-------------------------------------------- ^
compilation aborted for mod_esmf_ocn_roms.F90 (code 1)
